R/Design-class.R

```diff
@@ -409,110 +409,118 @@ DualResponsesSamplesDesign <- function(eff_model,
+#' without DLE and efficacy samples. It contain all slots in
+#' [`RuleDesign`] and [`TDDesign`] class object
+#'
+#' @slot data the data set of [`DataDual`] class object
+#' @slot eff_model the pseudo efficacy model to be used, an object class of
+#' [`ModelEff`]
+#'
+#' @export
+#'
 .DualResponsesDesign <-
   setClass(
     Class = "DualResponsesDesign",
-    representation(
-      Effmodel = "ModelEff",
+    representation = list(
+      eff_model = "ModelEff",
       data = "DataDual"
     ),
-    prototype(
+    prototype = prototype(
       nextBest = .NextBestMaxGain(),
       data = DataDual(doseGrid = 1:2),
       startingDose = 1,
       model = .LogisticIndepBeta()
     ),
-    contains = list("TDDesign")
+    contains = "TDDesign"
   )
-validObject(.DualResponsesDesign())
-
-
-##' Initialization function for 'DualResponsesDesign"
-##' @param data please refer to \code{\linkS4class{DualResponsesDesign}} class object
-##' @param Effmodel please refer to \code{\linkS4class{DualResponsesDesign}} class object
-##' @param \dots additional arguments for \code{\link{TDDesign}}
-##' @return the \code{\linkS4class{DualResponsesDesign}} class object
-##'
-##' @export
-##' @keywords methods
-DualResponsesDesign <- function(Effmodel,
+
+## constructor ----
+
+#' @rdname DualResponsesDesign-class
+#' @param data please refer to [`DualResponsesDesign`] class object
+#' @param eff_model please refer to [`DualResponsesDesign`] class object
+#' @param \dots additional arguments for \code{\link{TDDesign}}
+#' @inheritDotParams RuleDesign
+#'
+#' @example examples/design-class-DualResponsesDesign.R
+#' @export
+#'
+DualResponsesDesign <- function(eff_model,
                                 data,
                                 ...) {
   start <- TDDesign(data = data, ...)
   .DualResponsesDesign(start,
-    Effmodel = Effmodel,
+    eff_model = eff_model,
     data = data
   )
 }
 
-## ===============================================================================
-
-
-##' Class for the time-to-DLT augmented CRM design
-##'
-##' This class has special requirements for the \code{model} and \code{data}
-##' slots in comparison to the parent class \code{\linkS4class{Design}}:
-##'
-##' @slot model the model to be used, an object of or inheriting from class
-##' \code{\linkS4class{GeneralModel}}, see in particular
-##' \code{\linkS4class{DALogisticLogNormal}} and
-##' \code{\linkS4class{TITELogisticLogNormal}} which make use of the
-##' time-to-DLT data
-##' @slot data what is the dose grid, any previous data, etc., contained
-##' in an object of class \code{\linkS4class{DataDA}}
-##' @slot safetyWindow still to be documented.
-##'
-##' @example examples/design-class-DADesign.R
-##' @export
-##' @keywords classes
+
+# DADesign ----
+
+## class ----
+
+#' `DADesign`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' This class has special requirements for the \code{model} and \code{data}
```

Comment:
```suggestion
#' This class has special requirements for the `model` and `data`
```
we want to use the markdown syntax